backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/FeaturedLibraries.kt

```diff
@@ -0,0 +1,161 @@
+import org.jetbrains.kotlin.konan.library.resolver.KonanLibraryResolveResult
+import org.jetbrains.kotlin.library.toUnresolvedLibraries
+
+internal fun Context.getExportedDependencies(): List<ModuleDescriptor> = getDescriptorsFromFeaturedLibraries((config.exportedLibraries + config.sourceLibraries).toSet())
+internal fun Context.getSourceLibraryDescriptors(): List<ModuleDescriptor> = getDescriptorsFromFeaturedLibraries(config.sourceLibraries.toSet())
+
+private fun Context.getDescriptorsFromFeaturedLibraries(featuredLibraries: Set<KonanLibrary>) =
+    moduleDescriptor.allDependencyModules.filter {
+        when (val origin = it.konanModuleOrigin) {
+            CurrentKonanModuleOrigin, SyntheticModulesOrigin -> false
+            is DeserializedKonanModuleOrigin -> origin.library in featuredLibraries
+        }
+    }
+
+internal fun getExportedLibraries(
+        configuration: CompilerConfiguration,
+        resolvedLibraries: KonanLibraryResolveResult,
+        resolver: SearchPathResolver,
+        report: Boolean
+): List<KonanLibrary> = getFeaturedLibraries(
+    configuration.getList(KonanConfigKeys.EXPORTED_LIBRARIES),
+    resolvedLibraries,
+    resolver,
+    if (report) FeaturedLibrariesReporter.forExportedLibraries(configuration) else FeaturedLibrariesReporter.Silent
+)
+
+internal fun getSourceLibraries(
+    configuration: CompilerConfiguration,
+    resolvedLibraries: KonanLibraryResolveResult,
+    resolver: SearchPathResolver
+): List<KonanLibrary> = getFeaturedLibraries(
+    configuration.getList(KonanConfigKeys.SOURCE_LIBRARIES),
+    resolvedLibraries,
+    resolver,
+    FeaturedLibrariesReporter.forSourceLibraries(configuration)
+)
+
+internal fun getCoveredLibraries(
+    configuration: CompilerConfiguration,
+    resolvedLibraries: KonanLibraryResolveResult,
+    resolver: SearchPathResolver
+): List<KonanLibrary> = getFeaturedLibraries(
+    configuration.getList(KonanConfigKeys.LIBRARIES_TO_COVER),
+    resolvedLibraries,
+    resolver,
+    FeaturedLibrariesReporter.forCoveredLibraries(configuration)
+)
+
+private sealed class FeaturedLibrariesReporter {
+
+    abstract fun reportIllegalKind(library: KonanLibrary)
+    abstract fun reportNotIncludedLibraries(includedLibraries: List<KonanLibrary>, remainingFeaturedLibraries: Set<File>)
+
+    object Silent: FeaturedLibrariesReporter() {
+        override fun reportIllegalKind(library: KonanLibrary) {}
+        override fun reportNotIncludedLibraries(includedLibraries: List<KonanLibrary>, remainingFeaturedLibraries: Set<File>) {}
+
+    }
+
+    abstract class BaseReporter(val configuration: CompilerConfiguration) : FeaturedLibrariesReporter() {
+        protected abstract fun illegalKindMessage(kind: String, libraryName: String): String
+        protected abstract fun notIncludedLibraryMessageTitle(): String
+
+        override fun reportIllegalKind(library: KonanLibrary) {
+            val kind = if (library.isInterop) "Interop" else "Default"
```

Comment:
Since it is not clear anymore that the library here is either interop or default, then this code should check `isDefault` too.